### PR TITLE
[ASCII-1158] Add e2e test for /agent/version and /agent/flare api routes

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

With https://github.com/DataDog/datadog-agent/pull/24045 we're refactoring the internal agent api endpoints, lets add some e2e tests around the endpoints to ensure nothing breaks.

### CODE REVIEW

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+func (endpointInfo *agentEndpointInfo) url() *url.URL {
+	return &url.URL{
+		Scheme: endpointInfo.scheme,
+		Host:   net.JoinHostPort("localhost", strconv.Itoa(endpointInfo.port)),
+		Path:   endpointInfo.endpoint,
+	}
+}
+
+func (endpointInfo *agentEndpointInfo) fetchCommand(authtoken string) string {
+	var data string
```
Comment (dustmop) [inline test/new-e2e/tests/agent-shared-components/api/api_test.go:54]:
```suggestion
	data := endpointInfo.data
	if data == "" {
		data = "{}"
	}
```

would be more idiomatic way of assigning a default value.

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	if len(endpointInfo.data) > 0 {
+		data = endpointInfo.data
+	} else {
+		data = "{}"
+	}
+
+	// -L: follow redirects
```
Comment (dustmop) [inline test/new-e2e/tests/agent-shared-components/api/api_test.go:61]: The `-L` option isn't being used, but `-X` is.

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		authtoken,
+		endpointInfo.method,
+		endpointInfo.url().String(),
+		data,
+	)
+}
+
+func (v *apiSuite) TestInternalAgentAPIEndpoints() {
+	testcases := []struct {
+		endpoint agentEndpointInfo
```
Comment (dustmop) [inline test/new-e2e/tests/agent-shared-components/api/api_test.go:77]:
```suggestion
	agentEndpointInfo
```

If you embed the `agentEndpointInfo` instead, you can avoid a level of nesting for each of the test cases.

**Review by dustmop — APPROVED** [review]
Approved, with a few suggestions

File: test/new-e2e/tests/agent-shared-components/api/api_test.go (hunk 3 above)
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/24616/commits/f40c2fbe7dc338ccdb382c1a2fbc727e1842d84b

File: test/new-e2e/tests/agent-shared-components/api/api_test.go (hunk 2 above)
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/24616/commits/f40c2fbe7dc338ccdb382c1a2fbc727e1842d84b

File: test/new-e2e/tests/agent-shared-components/api/api_test.go (hunk 1 above)
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/24616/commits/f40c2fbe7dc338ccdb382c1a2fbc727e1842d84b

**Pull request merged**